### ISSUE TITLE
Create and revoke personal API tokens

### ISSUE THREAD

**Issue by amberrignell** [OpenFn/Lightning#147]
Allow users to issue and revoke personal API tokens from the user settings page

## Behaviour

![3dc328c2-24ab-4d4e-b7e3-88671b8880f1.jpeg](https://images.zenhubusercontent.com/55ba74804f7a08fbb7f40152/bbeb3585-c894-4c03-a196-e73db6db02a0)

1. When a user navigates to their user settings page (/profile), they should see a button that allows them to 'generate new API token'. 
2. When the button is clicked, the api token should appear in a table with the following columns on the same page: token, date issued. Each row should also have a dustbin icon delete button to delete the token and revoke access.
3. Beside each api token, there should be a button that allows them to delete the token. 

## Implementation Notes

- We currently do have a function for issuing "auth" tokens, they however are only valid for 30s.
- In this implementation tokens should last forever.
- We need to create a new kind of token, "api" - following on the implementation in `Accounts.generate_auth_token/1`.

## Questions

> @stuartc  what expiry date should we set?

We won't have an expiry as these tokens will be used indefinitely.

> @stuartc in other apps I've seen you get given the token to copy on generation but then you cannot see it again. How important is this? Let me know if I should create another issue to make sure we implement it down the line.

I think we should do this, but lets make it a different feature - when creating a new token I image displaying it in a modal with a message explaining it won't be visible again; and in the list we show the issue date and the first/last 5 characters for users to identify a token they have.

## For later

- Confirm the deletion of a token with a modal
- Show the token in full only once
- Hide the token and show only a partially identifiable number of characters
- Record the last time a token was used - and possibly the IP address associated as well
